### PR TITLE
dev: update data feed script

### LINKED ISSUE #44: dev: update data feed script
**Goal**

Write a simple CLI script in the `typescript/` folder that updates a data feed on a destination chain.
It gets the calldata from theoros either through a custom endpoint or a default one `https://theoros.pragma.build/`

Usage: `bun run update-data-feed--contract <contract_id> --feed-id <feed-id> --private-key <private-key>`

### PR DESCRIPTION
**Pull request by azurwastaken**
### Related Issue

Resolve #44

### CODE REVIEW

File: typescript/src/updater/update_feed.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+// ## Steps
+// 1. Theoros ? (choix de l'endpoint)
+// 2. Get calldata from Theoros
+// 3. Call `updateDataFeeds` avec la calldata
+// 4. Assertions - check that on the destination chain everything is correctly updated
+//    Show gas consumption
+
+import { Command } from "commander";
+import { ethers } from "ethers";
+import abi from "../../solidity/out/Pragma.sol/Pragma.json";
```
Comment (akhercha) [inline typescript/src/updater/update_feed.ts]: We should import this from the code and raise an error if it does not exists

File: typescript/src/updater/update_feed.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
+// ## Cli
```
Comment (akhercha) [inline typescript/src/updater/update_feed.ts]: This should be a separate package, not a single file, i.e:
```
typescript
├── pragma-deployer
└── scripts // <--- all the scripts here
     └── package.json etc
     └── src/update_feed.ts
```

File: typescript/src/updater/update_feed.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  try {
+    calldata = await getCalldataFromTheoros(theorosEndpoint, feedId);
+    console.log("Calldata retrieved from Theoros:", calldata);
+  } catch (error) {
+    console.error("Failed to retrieve calldata from Theoros:", error);
+    process.exit(1);
+  }
+
+  // 3. Call `updateDataFeeds` with the calldata
+  const { ethers } = require("ethers");
```
Comment (akhercha) [inline typescript/src/updater/update_feed.ts]: We don't want imports in the code here, should be top file

File: typescript/src/updater/update_feed.ts [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  const provider = new ethers.providers.JsonRpcProvider(process.env.RPC_URL);
+
+  const wallet = new ethers.Wallet(privateKey, provider);
+  const contract = new ethers.Contract(chainConfig.contractId, abi.abi, wallet);
+
+  try {
+    const tx = await contract.updateDataFeeds(calldata);
+    console.log("Transaction sent. Transaction hash:", tx.hash);
+
+    // 4. Assertions - check that everything is correctly updated on the destination chain
```
Comment (akhercha) [inline typescript/src/updater/update_feed.ts]: We should get the update on the chain & check that the timestamp is recent?

File: typescript/src/updater/update_feed.ts [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+  const config = chainConfigs[chainId];
+  if (!config) {
+    throw new Error(`Configuration non trouvée pour le chainId ${chainId}`);
+  }
+
+  return config;
+}
+
+function parseCommandLineArguments() {
```
Comment (akhercha) [inline typescript/src/updater/update_feed.ts]: nit: We should say in the function prototype that we return something - i.e. : 
```typescript
function parseCommandLineArguments(): returnTypeHere {
```

File: typescript/src/updater/update_feed.ts [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+import { Command } from "commander";
+import { ethers } from "ethers";
+import abi from "../../solidity/out/Pragma.sol/Pragma.json";
+
+interface ChainConfig {
+  contractId: string;
+}
+
+function getChainConfig(chainId: string): ChainConfig {
+  const chainConfigs: { [key: string]: ChainConfig } = {
```
Comment (akhercha) [inline typescript/src/updater/update_feed.ts]: This should be a configuration in a separate yaml file I think. And we just read the config there.

Maybe we can read directly from the `deployments` folder? And so if something isn't deploy we can't update it?

**Review by akhercha — CHANGES_REQUESTED** [review]
some comments

File: typescript/src/updater/update_feed.ts (hunk 1 above)
Comment (azurwastaken) [reply typescript/src/updater/update_feed.ts]: done

File: typescript/src/updater/update_feed.ts (hunk 2 above)
Comment (azurwastaken) [reply typescript/src/updater/update_feed.ts]: done

File: typescript/src/updater/update_feed.ts (hunk 3 above)
Comment (azurwastaken) [reply typescript/src/updater/update_feed.ts]: done

File: typescript/src/updater/update_feed.ts (hunk 5 above)
Comment (azurwastaken) [reply typescript/src/updater/update_feed.ts]: done

File: typescript/scripts/README.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
+# scripts
```
Comment (akhercha) [inline typescript/scripts/README.md:1]: Update this README, it should explain what this package is and what scripts are available

File: typescript/src/updater/update_feed.ts (hunk 6 above)
Comment (azurwastaken) [reply typescript/src/updater/update_feed.ts]: done i guess

File: typescript/scripts/README.md (hunk 7 above)
Comment (azurwastaken) [reply typescript/scripts/README.md:1]: done

File: typescript/src/updater/update_feed.ts (hunk 6 above)
Comment (akhercha) [reply typescript/src/updater/update_feed.ts]: Almost, the output deployment format is `deployments/[chain]/[oracle | dispatcher | pragma].json`

`pragma.json` will have the sol deployed contract like:
```json
{
"Pragma": "0x..."
}

Comment (azurwastaken) [reply typescript/src/updater/update_feed.ts]: done

**Review by EvolveArt — APPROVED** [review]
lgtm

**Pull request merged**